### PR TITLE
Implemented shutdown

### PR DESCRIPTION
**Pull request by g4s8**
#963 - stop accepting new claims when shutdown called and wait for all submited claims to be processed, then close the farm.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1549](https://github.com/zerocracy/farm/issues/1549) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1549](https://github.com/zerocracy/farm/issues/1549) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2161138173); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/zerocracy/claims/ClaimsRoutine.java [hunk 1]
```diff
@@ -148,4 +151,39 @@ public void run() {
...
+         */
+        ShutdownRunnable(final Runnable origin,
+            final ShutdownFarm.Hook shutdown) {
+            this.origin = origin;
+            this.shutdown = shutdown;
+        }
+
+        @Override
+        public void run() {
+            if (!this.shutdown.isStopping()) {
```
Comment (krzyk) [inline src/main/java/com/zerocracy/claims/ClaimsRoutine.java]: @g4s8 in a rare case you might end up in a STOPPED state and the run method will run.
I would add a check for stopping and stopped states.

File: src/main/java/com/zerocracy/shutdown/ShutdownFarm.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                }
+            }
+        }
+
+        /**
+         * Check shutdown in progress.
+         *
+         * @return TRUE if in progress
+         */
+        public boolean isStopping() {
```
Comment (krzyk) [inline src/main/java/com/zerocracy/shutdown/ShutdownFarm.java]: @g4s8 maybe a `stopping` (and accompanied by `stopped`)?

**Comment by g4s8** [comment]
@krzyk fixed, please look again

**Comment by krzyk** [comment]
@g4s8 thanks

**Comment by krzyk** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
The job [#1549](https://github.com/zerocracy/farm/issues/1549) is now out of scope

**Comment by 0crat** [comment]
Job [#1549](https://github.com/zerocracy/farm/issues/1549) is pending quality review, can't start a new order

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**